### PR TITLE
Setup continuous integration

### LINKED ISSUE #1: Setup continuous integration and deployment
- [x] Make sure Travis works fine - This will be taken care of in #16
- [x] Setup code coverage on the application (using coveralls)
- [x] Continuously deploy changes to Heroku
- [x] Enable codeclimate

### PR DESCRIPTION
**Pull request by diraulo**
Fixes #1

### Changes proposed by this PR

- Setup coveralls for code coverage
- Update `.travis.yml` to run `tests` rake task that generates merged coverage report
- disable email notifications
- setup continuous deployment of `master` branch to the Heroku app
- Add Travis + Coveralls badge to README
- Add PR template

### What did you learn working on this PR?
n/a

### Screenshots
no visual changes

### CODE REVIEW

**Comment by diraulo** [comment]
Make sure this is merged before you start working on anything else.

File: Gemfile [hunk 1]
```diff
@@ -21,11 +21,12 @@ group :development do
 end
 
 group :development, :test do
-  gem 'rspec-rails'
-  gem 'shoulda-matchers'
-  gem 'factory_girl_rails'
+  gem 'coveralls', require: false
```
Comment (faraznaeem) [inline Gemfile:24]: gems should be in alphabetical order raoul!
:)

File: Gemfile.lock [hunk 2]
```diff
@@ -271,4 +288,4 @@ RUBY VERSION
    ruby 2.4.0p0
 
 BUNDLED WITH
-   1.15.1
```
Comment (faraznaeem) [inline Gemfile.lock:274]: Why was the version changed?

File: Gemfile (hunk 1 above)
Comment (diraulo) [reply Gemfile:24]: And it is 😉 

The red stuff are things I removed, Green are the ones I added

File: coverage/.resultset.json [hunk 3]
```diff
@@ -0,0 +1,318 @@
+{
```
Comment (faraznaeem) [inline coverage/.resultset.json]: Should this file be added to git ignore?

File: lib/tasks/ci.rake [hunk 4]
```diff
@@ -1,5 +1,8 @@
+unless Rails.env.production?
+  require 'rspec/core/rake_task'
+  require 'cucumber/rake/task'
+  require 'coveralls/rake/task'
 
-require 'coveralls/rake/task'
```
Comment (faraznaeem) [inline lib/tasks/ci.rake]: great job Raoul! keep it up!

**Review by faraznaeem — APPROVED** [review]
Looks good, please ship it.

**Review by j0sv — APPROVED** [review]
I have no more comment. Approved. 
:shipit:

File: Gemfile.lock (hunk 2 above)
Comment (diraulo) [reply Gemfile.lock:274]: Whenever you run `bundle install` on your local machine if your version of bundler is greater than the one used when generating the `Gemfile.lock` it'll update this version number. :-)

**Pull request merged**